JS/Lesson19/salih/exercises.js

```diff
@@ -1,8 +1,157 @@
 // 1. Create a Person class with name, age, and gender properties, and a greet() method that logs a greeting message to the console. Then create a Student class that extends the Person class and adds a major property. Override the greet() method in the Student class to log a different greeting message to the console. Create 2 variables, one of type Person and one of type Student and log them to console.
+class Person{
+    constructor(name, age, gender){
+        this._name = name;
+        this._age = age;
+        this._gender = gender;
+    }
+    greet(){
+        console.log("Hello I am " + this._name);
+    }
+}
 
+class Student extends Person {
+    constructor(name, age, gender, major){
+        super(name, age, gender);
+        this._major = major;
+    }
+    greet(){
+        console.log("Hello I am " + this._name + ". My major is " + this._major);
+    }
+}
+
+const person1 = new Person("Salih", 25, "male");
+const student1 = new Student("Selin", 20, "female", "Computer Science");
+
+person1.greet();
+student1.greet();
+
+console.log("___________________________________");
 // 2. Create a Shape class with name and color properties, and a describe() method that logs a description of the shape to the console (e.g. "This is a blue square"). Then create a Square class that extends the Shape class and adds sideLength and area properties. Override the describe() method in the Square class to log a description of the square to the console (e.g. "This is a blue square with a side length of 5 and an area of 25").
 
+class Shape {
+    constructor(name, color){
+        this._name = name;
+        this._color = color;
+    }
+    describe(){
+        console.log("This is a " + this._color + " " + this._name);
+    }
+}
+
+class Square extends Shape{
+    constructor(name, color, sideLength, area){
```

Comment:
Square will always be a square, so you don't need to have a `name` parameter, you can instead hardcode it to "Square"